go/src/lib/lomipc/client_transport.go

```diff
@@ -0,0 +1,217 @@
+
+    LogInfo("Registered client (%s)", client)
+    return nil
+}
+
+
+func (tx *ClientTx) DeregisterClient() error {
+    defer func() {
+        tx.ClientRpc = nil
+        tx.ClientName = ""
+    }()
+
+    req := &LoMRequest { TypeDeregClient, tx.ClientName, tx.TimeoutSecs, MsgDeregClient{} }
+    reply := &LoMResponse{}
+    err := ClientCall(tx, "LoMTransport.SendToServer", req, reply)
+    if (err != nil) {
+        LogError("DeregisterClient: Failed to call SendToServer (%s) (%v)", tx.ClientName, err)
+        return err
+    }
+    if (reply.ResultCode != 0) {
+        return LogError("DeregisterClient: Server failed (%v) result(%d/%s)", tx.ClientName,
+                reply.ResultCode, reply.ResultStr)
+    }
+
+    res := reply.RespData
+    if x, ok := res.(MsgEmptyResp); !ok {
+        return LogError("DeegisterClient: Expect empty resp. (%T) (%v)", x, x)
+    }
+
+    LogInfo("Deregistered client (%s)", tx.ClientName)
+    return nil
+}
+
+
+func (tx *ClientTx) RegisterAction(action string) error {
+
+    req := &LoMRequest { TypeRegAction, tx.ClientName, tx.TimeoutSecs, MsgRegAction{action} }
+    reply := &LoMResponse{}
+    err := ClientCall(tx, "LoMTransport.SendToServer", req, reply)
+    if (err != nil) {
+        LogError("RegisterAction: Failed to call SendToServer (%s/%s) (%v)", tx.ClientName,
+                action, err)
+        return err
+    }
+    if (reply.ResultCode != 0) {
+        return LogError("RegisterAction: Server failed (%s/%s) result(%d/%s)", tx.ClientName,
+                action, reply.ResultCode, reply.ResultStr)
+    }
+
+    res := reply.RespData
+    if x, ok := res.(MsgEmptyResp); !ok {
+        return LogError("RegisterAction: Expect empty resp. (%T) (%v)", x, x)
+    }
+
+    LogInfo("Registered action (%s/%s)", tx.ClientName, action)
+    return nil
+}
+
+
+func (tx *ClientTx) DeregisterAction(action string) error {
+
+    req := &LoMRequest { TypeDeregAction, tx.ClientName, tx.TimeoutSecs, MsgDeregAction{action} }
+    reply := &LoMResponse{}
+    err := ClientCall(tx, "LoMTransport.SendToServer", req, reply)
+    if (err != nil) {
+        LogError("DeregisterAction: Failed to call SendToServer (%s/%s) (%v)", tx.ClientName,
+                action, err)
+        return err
+    }
+    if (reply.ResultCode != 0) {
+        return LogError("DeregisterAction: Server failed (%s/%s) result(%d/%s)", tx.ClientName,
+                action, reply.ResultCode, reply.ResultStr)
+    }
+
+    res := reply.RespData
+    if x, ok := res.(MsgEmptyResp); !ok {
+        return LogError("DeegisterAction: Expect empty resp. (%T) (%v)", x, x)
+    }
+
+    LogInfo("Deregistered action (%s/%s)", tx.ClientName, action)
+    return nil
+}
+
+func (tx *ClientTx) RecvServerRequest() (*ServerRequestData, error) {
+
+    req := &LoMRequest { TypeRecvServerRequest, tx.ClientName, tx.TimeoutSecs, MsgRecvServerRequest{} }
+    reply := &LoMResponse{}
+    err := ClientCall(tx, "LoMTransport.SendToServer", req, reply)
```

Comment:
Will both RecvServerRequest and SendServerResponse make calls to server ?

Reply:
yes